Error: unknown mnemonic `rep' -- `rep' on arrch64

hello !

this is very nice and useful project thank you for making it available.

Sadly it doesn't build on aarch64 architecture (mac M1) as it replies on some assembly instructions which are not supported.

e.g 

$ make
Scanning dependencies of target DisruptorStatic
[  2%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/BasicExecutor.cpp.o
[  4%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/BlockingWaitStrategy.cpp.o
[  6%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/BusySpinWaitStrategy.cpp.o
[  9%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/FixedSequenceGroup.cpp.o
[ 11%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/ProcessingSequenceBarrier.cpp.o
[ 13%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/ProducerType.cpp.o
[ 15%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/RoundRobinThreadAffinedTaskScheduler.cpp.o
[ 18%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/Sequence.cpp.o
[ 20%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/SequenceGroups.cpp.o
[ 22%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/SleepingWaitStrategy.cpp.o
[ 25%] Building CXX object Disruptor/CMakeFiles/DisruptorStatic.dir/SpinWait.cpp.o
/tmp/ccVC3BXq.s: Assembler messages:
/tmp/ccVC3BXq.s:119: Error: unknown mnemonic `rep' -- `rep'
/tmp/ccVC3BXq.s:220: Error: unknown mnemonic `rep' -- `rep'
/tmp/ccVC3BXq.s:241: Error: unknown mnemonic `rep' -- `rep'
make[2]: *** [Disruptor/CMakeFiles/DisruptorStatic.dir/build.make:193: Disruptor/CMakeFiles/DisruptorStatic.dir/SpinWait.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:152: Disruptor/CMakeFiles/DisruptorStatic.dir/all] Error 2
make: *** [Makefile:163: all] Error 2


I don't expect you to fix this as you obviously made this implementation for your use case with low latency in mind -  though may be you could still consider adding a similar fix as in this similar case?

https://github.com/nats-io/nats.c/issues/87

Regards, Roman

--- a/Disruptor/BuildConfig.h
+++ b/Disruptor/BuildConfig.h
@@ -35,7 +35,7 @@
 # if __arm__
 #  define DISRUPTOR_CPU_ARM
 #  define DISRUPTOR_CPU_ARM_32
-# #elif __aarch64__
+# elif __aarch64__
 #  define DISRUPTOR_CPU_ARM
 #  define DISRUPTOR_CPU_ARM_64
 # elif __x86_64__ || __ppc64__

